### PR TITLE
fix(security-tpm): Fix bug where TPM security providers cleared TPM's SRK and EK at constructor time

### PR DESCRIPTION
**Pull request by timtay-microsoft**
These security providers are expected to use the existing EK and SRK if they are available, and to create them only if it didn't exist previously. There was no reason to delete the existing EK and SRK every time.

The .NET SDK does not clear the SRK or EK at constructor time, for reference

I'm running this by a member of the TSS team since they more or less gave this code to us a while back, and they have a lot more context on this code.

#780 for additional context

### CODE REVIEW

File: iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/provisioning/setup/ProvisioningCommon.java [hunk 1]
```diff
@@ -239,7 +239,6 @@ public void tearDown()
 
         if (testInstance != null && testInstance.securityProvider != null && testInstance.securityProvider instanceof SecurityProviderTPMEmulator)
         {
-            testInstance.provisioningServiceClient = null;
```
Comment (timtay-microsoft) [inline iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/provisioning/setup/ProvisioningCommon.java:242]: Fixing a small test bug as well. No need to null this client when it is still used below for cleanup

File: provisioning/security/tpm-provider-emulator/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMEmulator.java [hunk 2]
```diff
@@ -66,8 +68,6 @@ public SecurityProviderTPMEmulator() throws SecurityProviderException
     {
         //SRS_SecurityProviderTPMEmulator_25_001: [ The constructor shall start the local TPM Simulator, clear persistent for EK and SRK if it exist, create persistent primary for EK and SRK. ]
         tpm = TpmFactory.localTpmSimulator();
-        clearPersistent(tpm, EK_PERSISTENT_HANDLE, "EK");
-        clearPersistent(tpm, SRK_PERSISTENT_HANDLE, "SRK");
```
Comment (abhipsaMisra) [inline provisioning/security/tpm-provider-emulator/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMEmulator.java:70]: q - is clearing the ek and srk ever relevant for a simulated tpm device (assuming it is a test device with a programmatically generated ek/ srk)?

File: provisioning/security/tpm-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMHsm.java [hunk 3]
```diff
@@ -135,6 +134,8 @@ private TPMT_PUBLIC createPersistentPrimary(Tpm tpm, TPM_HANDLE hPersistent, TPM
             throw new SecurityProviderException("Unexpected failure {" +  rc.name() + "} of TPM2_ReadPublic for {" + primaryRole + "}");
         }
 
+        log.info("Creating {} since the TPM didn't have one already", primaryRole);
```
Comment (abhipsaMisra) [inline provisioning/security/tpm-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMHsm.java:137]: Q - Wouldn't a hardware tpm device always come with an embedded ek? Is it even relevant for us to be creating an ek for an hsm device?

File: provisioning/security/tpm-provider-emulator/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMEmulator.java (hunk 2 above)
Comment (timtay-microsoft) [reply provisioning/security/tpm-provider-emulator/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMEmulator.java:70]: It likely isn't relevant for TPM simulators. The TPM simulator that we use has no EK or SRK on startup. Whether we attempt to clear them here or not should result in nothing done. With that said, we may as well remove this unnecessary code

File: provisioning/security/tpm-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMHsm.java (hunk 3 above)
Comment (timtay-microsoft) [reply provisioning/security/tpm-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMHsm.java:137]: In practice, yes, hardware TPMs are expected to have an EK. In theory though, it isn't required. I don't expect most users to hit this code path

Comment (timtay-microsoft) [reply provisioning/security/tpm-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMHsm.java:137]: To add additional context, TPMs support clearing EKs and replacing them with new EKs. That's why this code technically worked without these changes. It's just an unusual case.

Comment (abhipsaMisra) [reply provisioning/security/tpm-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMHsm.java:137]: That's true, I wouldn't expect this code path to be exercised with hsm flow; no harm in leaving it here though

**Review by abhipsaMisra — APPROVED** [review]

**Review by jamdavi — APPROVED** [review]

**Pull request merged**